patches/api/0403-More-vanilla-friendly-methods-to-update-trades.patch

```diff
@@ -0,0 +1,67 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sun, 16 Oct 2022 15:58:38 +0200
+Subject: [PATCH] More vanilla friendly methods to update trades
+
+
+diff --git a/src/main/java/org/bukkit/entity/Villager.java b/src/main/java/org/bukkit/entity/Villager.java
+index d61855b83da0776fe910dee8cde184c720571a71..efd1f31890627204f4718cd591b31e936da08196 100644
+--- a/src/main/java/org/bukkit/entity/Villager.java
++++ b/src/main/java/org/bukkit/entity/Villager.java
+@@ -60,8 +60,11 @@ public interface Villager extends AbstractVillager {
+      * A villager with a level of 1 and no experience is liable to lose its
+      * profession.
+      *
++     * This doesn't update the trades of this villager.
++     *
+      * @param level the new level
+      * @throws IllegalArgumentException if level not between [1, 5]
++     * @see #increaseLevel(int)
+      */
+     public void setVillagerLevel(int level);
+ 
+@@ -81,6 +84,28 @@ public interface Villager extends AbstractVillager {
+     public void setVillagerExperience(int experience);
+ 
+     // Paper start
++    /**
++     * Increases the level of this villager.
++     * This also unlock new recipes unlike the raw
++     * method {@link #setVillagerLevel(int)}.
++     * A villager with a level of 1 and no experience is liable to lose its
++     * profession.
++     * A master villager has a level of 5 in its profession and
++     * will unlock 10 trades (2 per level).
++     *
++     * @param amount The amount of level
++     * @return If this villager has at least earned one level
++     * @see #setVillagerLevel(int)
++     */
++    boolean increaseLevel(int amount);
++
++    /**
++     * Gives to this villager some potential new trades
++     * based to its profession and level.
++     * @param amount The amount of trades to give
++     */
++    void addTrades(int amount);
++
+     /**
+      * Gets the amount of times a villager has restocked their trades today
+      * @return The amount of trade restocks.
+diff --git a/src/main/java/org/bukkit/entity/WanderingTrader.java b/src/main/java/org/bukkit/entity/WanderingTrader.java
+index 9f25774659b29d8fcca3eb9d9487edff42dbad13..7226ce2a5b0aacf6dc70e7d748256c03e1d4af6f 100644
+--- a/src/main/java/org/bukkit/entity/WanderingTrader.java
++++ b/src/main/java/org/bukkit/entity/WanderingTrader.java
+@@ -74,5 +74,11 @@ public interface WanderingTrader extends AbstractVillager {
+      * @param location location to wander towards (world is ignored, will always use the entity's world)
+      */
+     void setWanderingTowards(@org.jetbrains.annotations.Nullable org.bukkit.Location location);
++
++    /**
++     * Gives to this wandering trader some potential new trades.
++     * @param amount The amount of trades to give
++     */
++    void addTrades(int amount);
```

Comment:
I don't think the same method should be in WanderingTrader. Due to the difference in how trades are added, a single method with a single amount doesn't really make sense to me. Some other scheme should be here to reflect the two separate pools of trades for wandering traders.